s3/s3.go

```diff
@@ -1008,6 +1016,15 @@ func (o *Object) Update(in io.Reader, src fs.ObjectInfo, options ...fs.OpenOptio
 		metaMtime: aws.String(swift.TimeToFloatString(modTime)),
 	}
 
+	if size > uploader.PartSize {
+		hash, err := src.Hash(fs.HashMD5)
+
+		if err == nil && matchMd5.MatchString(hash) {
```

Comment:
An encrypted file will neither report an error nor return a hash, so I check both here before using the hash.